pkg/appmanager/resourceConfig.go

```diff
@@ -1174,6 +1180,28 @@ func (rc *ResourceConfig) AddRuleToPolicy(
 	rc.SetPolicy(*policy)
 }
 
+func (rc *ResourceConfig) DeleteRuleFromPolicy(
+	policyName string,
+	rule *Rule,
+) {
+	// We currently have at most 1 policy, 'forwarding'
+	policy := rc.FindPolicy("forwarding")
+	if nil != policy {
+		for i, r := range policy.Rules {
+			if r.Name == rule.Name && r.FullURI == rule.FullURI {
+				// Remove old rule
+				if len(policy.Rules) == 1 {
+					rc.DeletePolicy(*policy)
```

Comment:
Enhancement: You can take advantage of this new DeletePolicy method here: https://github.com/F5Networks/k8s-bigip-ctlr/blob/master/pkg/appmanager/appManager.go#L1721